skimage/segmentation/trainable_segmentation.py

```diff
@@ -0,0 +1,151 @@
+from itertools import combinations_with_replacement
+import itertools
+import numpy as np
+from skimage import filters, feature
+from skimage import img_as_float32
+
+
+def _singlescale_basic_features(img, sigma, intensity=True, edges=True,
+                                texture=True):
+    """Features for a single value of the Gaussian blurring parameter ``sigma``
+    """
+    features = []
+    img_blur = filters.gaussian(img, sigma)
+    if intensity:
+        features.append(img_blur)
+    if edges:
+        features.append(filters.sobel(img_blur))
+    if texture:
+        H_elems = [
+            np.gradient(np.gradient(img_blur)[ax0], axis=ax1)
+            for ax0, ax1 in combinations_with_replacement(range(img.ndim), 2)
+        ]
+        eigvals = feature.hessian_matrix_eigvals(H_elems)
+        for eigval_mat in eigvals:
+            features.append(eigval_mat)
+    return features
+
+
+def _mutiscale_basic_features_singlechannel(
+    img, intensity=True, edges=True, texture=True, sigma_min=0.5, sigma_max=16
+):
+    """Features for a single channel image. ``img`` can be 2d or 3d.
```

Comment:
I think all the features you've made are nD?

Also, I would prefer passing in a sequence of sigmas, which would allow anisotropic sigmas, for example.

Reply:
I also think it's nD which is cool, but for the texture features I adapted the code of `filters.hessian` (not calling the function because I did not want to recompute the gaussian-filtered image since this where most of the time is spent) and the docstring of `filters.hessian` says `Defined only for 2-D and 3-D images`

Reply:
changed for the nD, I'll take a look at the anisotropic sigmas.